internal/events/target.go

```diff
@@ -0,0 +1,165 @@
+package events
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+type (
+	// A Listener is a function that listens for events of type T.
+	Listener[T any] func(T)
```

Comment:
the contract with this function is that we allow listener to run for some time, we do not expect it to return almost immediatelly, correct? 

if that's the case, do we maybe want to add context here for the sake of telemetry context propagation? 

And also add the context to Dispatch(), so that if you're sending an event that's part of some operation in telemetry terms, you could collect spans from listeners as well?

Reply:
We probably don't want a trace span that contains the entire execution stack of Pomerium, but sure I can add context to more places.